src/Files.App/Utils/Storage/StorageItems/BaseQueryResults.cs

```diff
@@ -45,12 +45,12 @@ public virtual IAsyncOperation<IReadOnlyList<IStorageItem>> GetItemsAsync()
 						{
 							if (colonSplit[0] == "System.FileName" || colonSplit[0] == "fileName" || colonSplit[0] == "name")
 							{
-								items = items.Where(x => Regex.IsMatch(x.Name, colonSplit[1].Replace("\"", "", StringComparison.Ordinal).Replace("*", "(.*?)", StringComparison.Ordinal), RegexOptions.IgnoreCase)).ToList();
+								items = items.Where(x => Regex.IsMatch(x.Name, Regex.Escape(colonSplit[1].Replace("\"", "", StringComparison.Ordinal)).Replace("\\*", ".*?"), RegexOptions.IgnoreCase)).ToList();
```

Comment:
@gumbarros Do you think we can utilize RegexHelper? If so, what the regex would look like?

Reply:
`RegexHelper` rely on compile-time strings for the source generator to work (because of the `GeneratedRegexAttribute`. Maybe you can use `RegexOptions.Compile` but I'm not familiar.

Reply:
Let's keep as is then, thanks!

Reply:
```suggestion
								items = items.Where(x => Regex.IsMatch(x.Name, Regex.Escape(colonSplit[1].Replace("\"", "", StringComparison.Ordinal)).Replace("\\*", ".*").Replace("\\?", "."), RegexOptions.IgnoreCase)).ToList();
```
* I don't think the regexp `?` is necessary for any string of zero or more characters.
* `?` should be replaced with the regexp `.` (any character).